sympy/physics/control/lti.py

```diff
@@ -853,20 +978,27 @@ class Parallel(Basic):
 
     """
     def __new__(cls, *args, evaluate=False):
-        if not all(isinstance(arg, (TransferFunction, Series, Parallel)) for arg in args):
+        if not all(isinstance(arg, (TransferFunction, TransferFunctionMatrix, Series, Parallel))
+            for arg in args):
             raise TypeError("Unsupported type of argument(s) for Parallel.")
 
         obj = super(Parallel, cls).__new__(cls, *args)
-        obj._var = None
-        for arg in args:
-            if obj._var is None:
-                obj._var = arg.var
-            elif obj._var != arg.var:
-                raise ValueError("All transfer functions should use the same complex"
-                    " variable of the Laplace transform.")
-        if evaluate:
-            return obj.doit()
-        return obj
+        obj.is_SISO = all(isinstance(arg.doit(), TransferFunction) for arg in args)
```

Comment:
`is_SISO` is a weird name.  A TFM with shape `(1, 1)` has a single input and output, but is not siso. I'd be inclined to either :

* give this a private name like `_is_not_matrix` (so that it can be removed later)
* create separate `ParallelMatrix` and `SeriesMatrix` classes - almost every method starts with `if is_SISO`, which is a big flag that you should have two different classes

Reply:
Separate classes might be a good idea, but I think it would be better to do that as a different patch. 
Also, (1,1) TFMatrix should be considered SISO.

Reply:
> Also, (1,1) TFMatrix should be considered SISO.

If this is the case, then all of the places where `is_SISO` is used need to be changed to check something else instead - because what they actually care about is "are any of my arguments `TFMatrix` instances"?

Reply:
In this case, I think it's fine to use `_is_not_matrix`. 

> create separate `ParallelMatrix` and `SeriesMatrix` classes - almost every method starts with `if is_SISO`, which is a big flag that you should have two different classes

I have had a look at other packages/libraries like python-control and Harold and what they have is a single transfer function object. Their algebraic code also does the same, that is, uses an [attribute](https://github.com/ilayn/harold/blob/517e57f119f05d0c10b1ff35f0b0e18fcdcfab39/harold/_classes.py#L658) to differentiate SISO and MIMO TF. I wanted to introduce two new classes for Transfer function matrix interconnection in the beginning, but we could also use these existing `Series` and `Parallel` classes for TFM interconnection, so I did that instead.